### PR TITLE
Add Sign In Modal test

### PR DESCRIPTION
**Pull request by dzhebeuilovo**
#20

### CODE REVIEW

File: app/components/signinModal/tests/index.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe('Signin Modal', () => {
+  const state = {
+    isModalOpen: true,
+    user: {
+      email: '',
+      password: ''
+    }
+  };
+  const signinModalComponent = mount(<SigninModal />);
+  signinModalComponent.setState(state);
```
Comment (maratfakhreev) [inline app/components/signinModal/tests/index.test.js]: move it after consts block

File: app/components/signinModal/tests/index.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  it('renders Modal with form fields', () => {
+    expect(signinModalComponent.find(Modal).length).toEqual(1);
+    expect(modalDialogContent.getElementsByClassName('form-group').length).toEqual(2);
+    expect(modalDialogContent.getElementsByClassName('form-control').length).toEqual(2);
+  });
+
+  it('input email has error class', () => {
+    const inputEmail = modalDialogContent.querySelector('[name=email]');
+    ReactTestUtils.Simulate.change(inputEmail, {
```
Comment (maratfakhreev) [inline app/components/signinModal/tests/index.test.js]: please add empty line

File: app/components/signinModal/tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const modalDialogContent = signinModalComponent.find(Modal).node._modal.getDialogElement();
+
+  it('renders Modal with form fields', () => {
+    expect(signinModalComponent.find(Modal).length).toEqual(1);
+    expect(modalDialogContent.getElementsByClassName('form-group').length).toEqual(2);
+    expect(modalDialogContent.getElementsByClassName('form-control').length).toEqual(2);
+  });
+
+  it('input email has error class', () => {
+    const inputEmail = modalDialogContent.querySelector('[name=email]');
```
Comment (maratfakhreev) [inline app/components/signinModal/tests/index.test.js]: please add empty line

File: app/components/signinModal/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        name: 'email',
+        value: 'short'
+      }
+    });
+
+    expect(inputEmail.parentElement.classList.contains('has-error')).toEqual(true);
+  });
+
+  it('input email has success class', () => {
+    const inputEmail = modalDialogContent.querySelector('[name=email]');
```
Comment (maratfakhreev) [inline app/components/signinModal/tests/index.test.js]: please add empty line

File: app/components/signinModal/tests/index.test.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const signinModalComponent = mount(<SigninModal />);
+  signinModalComponent.setState(state);
+  const modalDialogContent = signinModalComponent.find(Modal).node._modal.getDialogElement();
+
+  it('renders Modal with form fields', () => {
+    expect(signinModalComponent.find(Modal).length).toEqual(1);
+    expect(modalDialogContent.getElementsByClassName('form-group').length).toEqual(2);
+    expect(modalDialogContent.getElementsByClassName('form-control').length).toEqual(2);
+  });
+
```
Comment (maratfakhreev) [inline app/components/signinModal/tests/index.test.js:24]: add `context` to clearify specs semantic. Like:

context(‘when email is changed’, () => {
    it(‘it includes error class’, () => {

File: app/components/signinModal/tests/index.test.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  it('renders Modal with form fields', () => {
+    expect(signinModalComponent.find(Modal).length).toEqual(1);
+    expect(modalDialogContent.getElementsByClassName('form-group').length).toEqual(2);
+    expect(modalDialogContent.getElementsByClassName('form-control').length).toEqual(2);
+  });
+
+  describe('when email is empty', () => {
+    it('include error class', () => {
```
Comment (EmilShakirov) [inline app/components/signinModal/tests/index.test.js]: typo, should be **includes**

**Comment by maratfakhreev** [comment]
🚬:horse:

**Comment by EmilShakirov** [comment]
Could you please also add test for function calls? Example can be [here](https://github.com/fs/react-base/blob/master/app/components/navigation/tests/right.test.js#L54)

**Comment by dzhebeuilovo** [comment]
@EmilShakirov ok

**Comment by maratfakhreev** [comment]
🆗

**Pull request merged**